pkg/auth/dependency/interaction/infos.go

```diff
@@ -0,0 +1,58 @@
+package interaction
+
+type IdentityInfo struct {
+	ID     string                 `json:"id"`
+	Type   IdentityType           `json:"type"`
+	Claims map[string]interface{} `json:"claims"`
+}
+
+type IdentityType string
+
+const (
+	IdentityTypeLoginID IdentityType = "login_id"
+	IdentityTypeOAuth   IdentityType = "oauth"
+)
+
+const (
+	// IdentityClaimOAuthProvider is a claim with a map value like `{ "type": "azureadv2", "tenant": "test" }`.
+	IdentityClaimOAuthProvider string = "https://auth.skygear.io/claims/oauth/provider"
+	// IdentityClaimOAuthSubjectID is a claim with a string value like `1098765432`.
+	IdentityClaimOAuthSubjectID string = "https://auth.skygear.io/claims/oauth/subject-id"
+	// IdentityClaimOAuthData is a claim with a map value containing raw OAuth provider profile.
+	IdentityClaimOAuthProfile string = "https://auth.skygear.io/claims/oauth/profile"
+
+	// IdentityClaimLoginIDUniqueKey is a claim with a string value containing the unique normalized login ID.
+	IdentityClaimLoginIDUniqueKey string = "https://auth.skygear.io/claims/login_id/unique-key"
+)
+
+type AuthenticatorInfo struct {
+	ID     string                 `json:"id"`
+	Type   AuthenticatorType      `json:"type"`
+	Secret string                 `json:"secret"`
+	Props  map[string]interface{} `json:"props"`
+}
+
+type AuthenticatorType string
+
+const (
+	AuthenticatorTypePassword     AuthenticatorType = "password"
+	AuthenticatorTypeOTP          AuthenticatorType = "otp"
+	AuthenticatorTypeBearerToken  AuthenticatorType = "bearer_token"
+	AuthenticatorTypeRecoveryCode AuthenticatorType = "recovery_code"
+)
+
+const (
+	// AuthenticatorPropTOTPDisplayName is a claim with string value for TOTP display name.
+	AuthenticatorPropTOTPDisplayName string = "https://auth.skygear.io/claims/otp/display-name"
+
+	// AuthenticatorPropOOBChannelType is a claim with string value for OOB OTP channel type.
+	AuthenticatorPropOOBChannelType string = "https://auth.skygear.io/claims/otp/channel-type"
```

Comment:
`/oob/channel-type`?

`https://auth.skygear.io/claims/login_id/unique-key`: mixed `_` and `-`.

`https://auth.skygear.io/claims/bearer-token/parent-id`: `bearer-token` vs `bearer_token` (AuthenticatorTypeRecoveryCode).